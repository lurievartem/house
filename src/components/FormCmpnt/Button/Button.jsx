import React, { PropTypes } from 'react';

const Button = ({ onClick, type, text, className, ...rest }) => (
    <button
        type={type}
        className={className}
        onClick={onClick}
        {...rest}
    >
        {text}
    </button>
);

Button.propTypes = {
    onClick: PropTypes.func,
    type: PropTypes.string,
    text: PropTypes.string,
    className: PropTypes.string,
};

Button.defaultProps = {
    className: 'btn'
};

export default Button;
