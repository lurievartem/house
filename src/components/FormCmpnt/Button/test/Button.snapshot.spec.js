import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

it('Button snapshot', () => {
    const btn = renderer.create(
        <Button type="submit" text="submit" className="btn" />
    ).toJSON();

    expect(btn).toMatchSnapshot();
});
