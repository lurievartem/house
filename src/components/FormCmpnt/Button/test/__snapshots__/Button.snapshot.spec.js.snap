exports[`test Button snapshot 1`] = `
<button
  className="btn"
  onClick={undefined}
  type="submit">
  submit
</button>
`;
